homeassistant/components/bmw_connected_drive/sensor.py

```diff
@@ -521,66 +173,45 @@ def __init__(
 
     def update(self) -> None:
         """Read new state data from the library."""
-        _LOGGER.debug("Updating %s", self._vehicle.name)
-        vehicle_state = self._vehicle.state
+        _LOGGER.debug("Updating sensors of %s", self._vehicle.name)
+        vehicle_state = self._vehicle.status
         sensor_key = self.entity_description.key
+        sensor_value = None
+
         if sensor_key == "charging_status":
-            self._attr_native_value = getattr(vehicle_state, sensor_key).value
-        elif self.unit_of_measurement == VOLUME_GALLONS:
-            value = getattr(vehicle_state, sensor_key)
-            value_converted = self.hass.config.units.volume(value, VOLUME_LITERS)
-            self._attr_native_value = round(value_converted)
-        elif self.unit_of_measurement == LENGTH_MILES:
-            value = getattr(vehicle_state, sensor_key)
-            value_converted = self.hass.config.units.length(value, LENGTH_KILOMETERS)
-            self._attr_native_value = round(value_converted)
+            sensor_value = getattr(vehicle_state, sensor_key).value
         elif self._service is None:
-            self._attr_native_value = getattr(vehicle_state, sensor_key)
-        elif self._service == SERVICE_LAST_TRIP:
-            vehicle_last_trip = self._vehicle.state.last_trip
-            if sensor_key == "date_utc":
-                date_str = getattr(vehicle_last_trip, "date")
-                if parsed_date := dt_util.parse_datetime(date_str):
-                    self._attr_native_value = parsed_date.isoformat()
-                else:
-                    _LOGGER.debug(
-                        "Could not parse date string for 'date_utc' sensor: %s",
-                        date_str,
+            sensor_value = getattr(vehicle_state, sensor_key)
+
+            if isinstance(sensor_value, tuple):
```

Comment:
Side note: It would be nice if we can define a data lookup and formatting function in the descriptions so we don't need all if/else checks here.

Example:
https://github.com/home-assistant/core/blob/59f10373ada8a784c2bd3cf0e8bfa91821baca80/homeassistant/components/airly/sensor.py#L120

Reply:
Please check if this is the implementation you were thinking of!